.github/workflows/ci.yml

```diff
@@ -45,6 +45,12 @@ jobs:
         uses: actions/setup-dotnet@v1
         with:
           dotnet-version: 3.1.404
+      # WORKAROUND: Reset NuGet sources to default to avoid restore issues
```

Comment:
It seems that `setup-dotnet` is somehow broken  https://github.com/actions/setup-dotnet/issues/155). 
The workaround that I would suggest we choose is to add an extra step to clear the http-cache. I've tried this and it succeeded 5 times in a row:
```
# Workaround for https://github.com/actions/setup-dotnet/issues/155
- name: Consul | AspNetCore | Clear http-cache
  run: dotnet nuget list source && dotnet nuget locals http-cache --clear
```